### ISSUE TITLE
Question about AVB implementation

### ISSUE THREAD

**Issue by stefanwebb** [wiseodd/generative-models#13]
Hi, thanks for a useful library!

I notice in the AVB implementation that the discriminator is only a function of z, rather than also one of x. Also, the real and fake samples are swapped in the discriminator loss. Is this on purpose?

In my implementation of AVB, I do exactly as the paper says in Algorithm 1 but I find it's unstable like a vanilla GAN. I have to slow down both learning rates, or make the discriminator Eq (3.3) train slower than the ELBO bound, Eq (3.9)

**Comment by wiseodd** [comment]
Hi,

You're right. It's a mix-up with AAE implementation.
I've updated it via 02f95fcbb26829cafb698baaf8040bf1d495bb4a.
Thanks.

One thing though, in Algorithm 1, it says that we need to minimize T, whereas in 
Eq. 3.3, it says max. I found that to make it works, it should be max.

**Comment by stefanwebb** [comment]
That looks okay now but there is another serious bug

The disc and loglike terms have different scaling from the previous tf.reduce_mean()'s in [this line](https://github.com/wiseodd/generative-models/blob/master/VAE/adversarial_vb/avb_tensorflow.py#L114)

The term disc is scaled by 100 whereas the loglike term is scaled by 100 * 784

This is a problem because it affects their relative contribution to the gradient of the elbo

One solution would be to change [this line](https://github.com/wiseodd/generative-models/blob/master/VAE/adversarial_vb/avb_tensorflow.py#L112) to ") * 784"

This bug may be present in other algorithms in the library as well

Fixing it up it seems like it's converging much much faster to sensible samples

**Comment by wiseodd** [comment]
Thanks!
Fixed in 300bda9e1a0a37fd9cb9d6d3dede9e5fb22ca153

Checked the other algos as well, but they should be fine.
